### PR TITLE
IGNITE-13895: SSL usage in ducktape tests

### PR DESCRIPTION
**Pull request by Sega76**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/tests/certs/functions.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    if [[ ${ALIAS} == "" ]] || [[ ${DNAME} == "" ]] || [[ ${PSWD} == "" ]]
+    then
+        error "makeRoot: Need ALIAS, DNAME, PSWD"
+    fi
+
+    rm ${ALIAS}.jks
+    rm ${ALIAS}.pem
+
+    keytool -genkeypair -keystore ${ALIAS}.jks -alias ${ALIAS} -dname "${DNAME}" -ext bc:c -storepass ${PSWD} \
```
Comment (timoninmaxim) [inline modules/ducktests/tests/certs/functions.sh:16]: Please check bash checkstyle with Inteliij IDEA plugin. It will write down all possible issues. At least ${PSWD} must be in quotes, as it can contain special symbols like &.

File: modules/ducktests/tests/docker/run_tests.sh [hunk 2]
```diff
@@ -158,5 +158,7 @@ if [[ -n "$MAX_PARALLEL" ]]; then
   DUCKTAPE_OPTIONS="$DUCKTAPE_OPTIONS --max-parallel $MAX_PARALLEL"
 fi
 
+"$SCRIPT_DIR"/../certs/mkcerts.sh
```
Comment (timoninmaxim) [inline modules/ducktests/tests/docker/run_tests.sh:161]: This is mechanism for docker env. How is it going to be distributed on real nodes on Jenkins?

File: modules/ducktests/tests/ignitetest/services/utils/control_utility.py [hunk 3]
```diff
@@ -31,9 +31,17 @@ class ControlUtility:
     """
     BASE_COMMAND = "control.sh"
 
-    def __init__(self, cluster, text_context):
+    def __init__(self, cluster,
+                 key_store_jks: str = None, key_store_pwd: str = "123456",
```
Comment (timoninmaxim) [inline modules/ducktests/tests/ignitetest/services/utils/control_utility.py:35]: Password 123456 is a magic variable. Are there any alternatives? Config, global variable, param, or smth else?
At least setting global variable in run_tests.sh looks better.

File: modules/ducktests/tests/ignitetest/tests/ssl_test.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+from ignitetest.utils.version import DEV_BRANCH, IgniteVersion, LATEST_2_9, LATEST_2_8
+
+
+# pylint: disable=W0223
+class SslTest(IgniteTest):
+    """
+    Ssl test.
+    """
+    @cluster(num_nodes=3)
+    @ignite_versions(str(DEV_BRANCH), str(LATEST_2_9), str(LATEST_2_8))
```
Comment (timoninmaxim) [inline modules/ducktests/tests/ignitetest/tests/ssl_test.py:36]: Actually do we need check of LATEST versions?

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/ducktests/tests/certs/functions.sh (hunk 1 above)
Comment (Sega76) [reply modules/ducktests/tests/certs/functions.sh]: fixed

File: modules/ducktests/tests/docker/run_tests.sh (hunk 2 above)
Comment (Sega76) [reply modules/ducktests/tests/docker/run_tests.sh:161]: first certificates are generated, then copied to the nodes

File: modules/ducktests/tests/ignitetest/services/utils/control_utility.py (hunk 3 above)
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/control_utility.py]: fixed

File: modules/ducktests/tests/ignitetest/tests/ssl_test.py (hunk 4 above)
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/tests/ssl_test.py:36]: I think yes, these versions are now used

**Review by timoninmaxim — APPROVED** [review]

**Review by ivandasch — APPROVED** [review]

**Review by map7000 — APPROVED** [review]

File: modules/ducktests/tests/ignitetest/services/ignite_app.py [hunk 5]
```diff
@@ -16,14 +16,14 @@
 """
 This module contains the base class to build Ignite aware application written on java.
 """
-
```
Comment (nizhikov) [inline modules/ducktests/tests/ignitetest/services/ignite_app.py:19]: Why this change?
